### PR TITLE
prompting: add `snap-prompting-control` interface follow-up

### PR DESCRIPTION
**Pull request by olivercalder**
This is a follow-up to PR #12, which I merged prematurely. A few additional fixes are required.

The `snap-prompting-control` interface needs tests in order to pass tests. It should be very similar to the `snap-themes-control` interface.

### CODE REVIEW

File: daemon/api_prompting.go [hunk 1]
```diff
@@ -44,9 +41,6 @@ var (
 		POST:        postRequest,
 		ReadAccess:  interfaceOpenAccess{Interface: "snap-prompting-control"},
```
Comment (olivercalder) [inline daemon/api_prompting.go:42]: Here

File: daemon/api_prompting.go [hunk 2]
```diff
@@ -33,9 +33,6 @@ var (
 		Path:       "/v2/prompting/requests",
 		GET:        getRequests,
 		ReadAccess: interfaceOpenAccess{Interface: "snap-prompting-control"},
```
Comment (olivercalder) [inline daemon/api_prompting.go:35]: Here

File: daemon/api_access_control.go [hunk 3]
```diff
@@ -34,21 +34,15 @@ var (
 		GET:         getRules,
 		POST:        postRules,
 		ReadAccess:  interfaceOpenAccess{Interface: "snap-prompting-control"},
```
Comment (olivercalder) [inline daemon/api_access_control.go:36]: Here

File: daemon/api_access_control.go [hunk 4]
```diff
@@ -34,21 +34,15 @@ var (
...
-		// Need to add polkitActionPrompting to daemon/api.go and register
-		// prompt UI clients with it.
+		WriteAccess: interfaceAuthenticatedAccess{Interface: "snap-prompting-control", Polkit: polkitActionManage},
 	}
 
 	accessRuleCmd = &Command{
 		Path:        "/v2/access-control/rules/{id}",
 		GET:         getRule,
 		POST:        postRule,
 		ReadAccess:  interfaceOpenAccess{Interface: "snap-prompting-control"},
```
Comment (olivercalder) [inline daemon/api_access_control.go:44]: Here

**Review by olivercalder — COMMENTED** [review]
I think we should have read access behind `interfaceAuthenticatedAccess{..., Polkit: polkitActionManage}` as well, otherwise prompt clients could leak information about the access activity of other users.

I've noted the lines of interest below. @robert-ancell what do you think about this?

**Comment by robert-ancell** [comment]
> I've noted the lines of interest below. @robert-ancell what do you think about this?

I'm not an expert on this part of snapd but it looks correct to me. @jhenstridge this might be good for you to check?

**Pull request merged**